File: src/main/java/db/stats/GameStageDeltaDAOImpl.java
package db.stats;

import db.DBHelper;
import model.stats.CreepsPerMin;
import model.stats.GameStageDelta;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.logging.Level;
import java.util.logging.Logger;

/**
 * @author Josiah Kendall
 */
public class GameStageDeltaDAOImpl implements GameStageDeltasDAO{

    private DBHelper dbHelper;

    public GameStageDeltaDAOImpl(DBHelper dbHelper) {
        this.dbHelper = dbHelper;
    }

    @Override
    public int saveDeltas(GameStageDelta gameStageDelta, String tableName) {
        try {
            String queryString = String.format(
                    "Insert into %s ("
                            + "ZeroToTen,"
                            + "TenToTwenty,"
                            + "TwentyToThirty,"
                            + "ThirtyToEnd) values (%s, %s, %s, %s)",
                    tableName,
                    gameStageDelta.getZeroToTen(),
                    gameStageDelta.getTenToTwenty(),
                    gameStageDelta.getTwentyToThirty(),
                    gameStageDelta.getThirtyToEnd()
            );
            return dbHelper.ExecuteSqlScript(queryString);
        } catch (SQLException | IllegalStateException | NullPointerException ex) {
            Logger.getLogger(CreepsPerMinDAOImpl.class.getName()).log(Level.SEVERE, null, ex);
        }

        return -1;
    }

    @Override
    public void deleteDeltas(long gameStageDeltasId, String tableName) {

    }

    @Override
    public GameStageDelta getGameStageDeltas(long gameStageDeltasId, String tableName) {
        try {
            // #sorrynotsorry
            String query = "SELECT * from "+tableName+" WHERE Id = " + gameStageDeltasId;
            ResultSet resultSet = dbHelper.ExecuteSqlQuery(query);
            if (resultSet.next()) {
                GameStageDelta gameStageDelta = new GameStageDelta();
                gameStageDelta.setId(gameStageDeltasId);
                gameStageDelta.setZeroToTen(resultSet.getDouble("ZeroToTen"));
                gameStageDelta.setTenToTwenty(resultSet.getDouble("TenToTwenty"));
                gameStageDelta.setTwentyToThirty(resultSet.getDouble("TwentyToThirty"));
                gameStageDelta.setThirtyToEnd(resultSet.getDouble("ThirtyToEnd"));

                return gameStageDelta;
            }
        } catch (SQLException | IllegalStateException ex) {
            Logger.getLogger(CreepsPerMinDAOImpl.class.getName()).log(Level.SEVERE, null, ex);
        }

        return null;
    }
}

File: src/main/java/db/stats/GameStageDeltaDAOImpl.kt
package db.stats

import db.DBHelper
import extensions.produceGameStageStat
import model.stats.GameStageDelta

import java.sql.SQLException
import java.util.logging.Level
import java.util.logging.Logger

/**
 * @author Josiah Kendall
 */
class GameStageDeltaDAOImpl(private val dbHelper: DBHelper) : GameStageDeltasDAO {

    override fun saveDeltas(gameStageDelta: GameStageDelta, tableName: String, timelineId: Long): Long {
        try {
            val queryString = String.format(
                    "Insert into %s ("
                            + "timelineId,"
                            + "ZeroToTen,"
                            + "TenToTwenty,"
                            + "TwentyToThirty,"
                            + "ThirtyToEnd) values (%s, %s, %s, %s, %s)",
                    tableName,
                    timelineId,
                    gameStageDelta.zeroToTen,
                    gameStageDelta.tenToTwenty,
                    gameStageDelta.twentyToThirty,
                    gameStageDelta.thirtyToEnd
            )
            return dbHelper.executeSQLScript(queryString)
        } catch (ex: SQLException) {
            Logger.getLogger(CreepsPerMinDAOImpl::class.java.name).log(Level.SEVERE, null, ex)
        } catch (ex: IllegalStateException) {
            Logger.getLogger(CreepsPerMinDAOImpl::class.java.name).log(Level.SEVERE, null, ex)
        } catch (ex: NullPointerException) {
            Logger.getLogger(CreepsPerMinDAOImpl::class.java.name).log(Level.SEVERE, null, ex)
        }

        return -1
    }

    override fun deleteDeltas(gameStageDeltasId: Long, tableName: String) {

    }

    fun getGameStageDeltasForTimeline(timelineId: Long, tableName: String) : GameStageDelta {
        val sql = "SELECT * FROM $tableName " +
                "WHERE timelineId = $timelineId"
        val result = dbHelper.executeSqlQuery(sql)
        result.next()
        return result.produceGameStageStat()
    }

    override fun getGameStageDeltas(gameStageDeltasId: Long, tableName: String): GameStageDelta? {
        try {

            val query = "SELECT * from $tableName WHERE Id = $gameStageDeltasId"
            val resultSet = dbHelper.executeSqlQuery(query)
            if (resultSet.next()) {
                val gameStageDelta = GameStageDelta()
                gameStageDelta.id = gameStageDeltasId
                gameStageDelta.zeroToTen = resultSet.getDouble("ZeroToTen")
                gameStageDelta.tenToTwenty = resultSet.getDouble("TenToTwenty")
                gameStageDelta.twentyToThirty = resultSet.getDouble("TwentyToThirty")
                gameStageDelta.thirtyToEnd = resultSet.getDouble("ThirtyToEnd")

                return gameStageDelta
            }
        } catch (ex: SQLException) {
            Logger.getLogger(CreepsPerMinDAOImpl::class.java.name).log(Level.SEVERE, null, ex)
        } catch (ex: IllegalStateException) {
            Logger.getLogger(CreepsPerMinDAOImpl::class.java.name).log(Level.SEVERE, null, ex)
        }

        return null
    }
}
